### PR TITLE
tweaking Example2

### PR DESCRIPTION
**Pull request by achubaty**
1. use sentence-based line wrapping:
    a) it's way easier to edit without long lines;
    b) plays nicer with git and viewing changes.

2. removed the copy/paste of the module code and instead parse the module code directly, which will ensure changes propagate to the Rmd.

3. fixed `SpaDES.core` version requirement in `projectSpeciesDist` module

@CeresBarros TODO: track the pre-built figures in git

### CODE REVIEW

File: Example2_SDMMaxent.Rmd [hunk 1]
```diff
@@ -25,27 +25,43 @@ knitr::opts_chunk$set(cache = 2, cache.rebuild = FALSE)
...
+In this example this script is called `Example2_SDMMaxent.R`.
+The script begins with a few lines of code that ensure necessary packages are installed and loaded (see [Reproducible package installation]).
+It then defines the necessary folder directories for the simulation and creates the modules in the `modules/` folder:
 
 ```{r globalscript, eval = FALSE}
 ## Get necessary R packages, but don't load any
 if (!require("Require")) install.packages("Require")
 Require::Require("PredictiveEcology/SpaDES.install@development")
 SpaDES.install::installSpaDES(ask = TRUE)
+Require("PredictiveEcology/SpaDES.core@development")
```
Comment (CeresBarros) [inline Example2_SDMMaxent.Rmd]: Please don't load packages yet. It messes up with `SpaDES.install::makeSureAllPackagesInstalled`
SpaDES is loaded after `SpaDES.install::makeSureAllPackagesInstalled` right before setting up the simulation

File: Example2_SDMMaxent.Rmd [hunk 2]
```diff
@@ -85,436 +101,64 @@ SpaDES.core::setPaths(cachePath = file.path(mainDir, "cache"),
...
-  
-  if (!suppliedElsewhere(sim$studyAreaRas)) {
-    ## code check: did the user supply a study area?
-    stop("Please supply a 'studyAreaRas' SpatRaster")
-  }
-  
-  # ! ----- STOP EDITING ----- ! #
-  return(invisible(sim))
-}
+```{r dataModule1, code = readLines(file.path("modules", "speciesAbundanceData", "speciesAbundanceData.R")), eval = FALSE}
```
Comment (CeresBarros) [inline Example2_SDMMaxent.Rmd:141]: brilliant, thanks!

**Review by CeresBarros — CHANGES_REQUESTED** [review]

**Comment by CeresBarros** [comment]
Thanks @achubaty , could you please just remove the `Require` line with `SpaDES.core`  after `SpaDES.install`?

File: Example2_SDMMaxent.Rmd (hunk 1 above)
Comment (achubaty) [reply Example2_SDMMaxent.Rmd]: yup, i should have done `require = FALSE`, but i removed this line

File: Example2_SDMMaxent.Rmd (hunk 2 above)
Comment (achubaty) [reply Example2_SDMMaxent.Rmd:141]: *bows*

**Comment by CeresBarros** [comment]
Thanks :)

**Pull request merged**